File: 1013/识别三角形.cpp
#include <iostream>
using namespace std;
int main()
{
	int a,b,c;
	cin>>a>>b>>c;
	if((a+b)<c)
	cout<<"NO";
	else if((a+c)<b)
		 cout<<"NO";
		 else if((b+c)<a)
		      cout<<"NO";
		      else if(a==b,b==c)
		           cout<<"Equilateral";
		           else if((a*a+b*b)==c*c)
		                cout<<"Right";
		                else if(a*a==(b*b+c*c))
		                     cout<<"Right";
		                     else if((a*a+c*c)==b*b)
		                          cout<<"Right";
		                          else cout<<"General";
	return 0;
}
		

File: 1013/识别三角形.pas
var
 n,m,j,k,l:longint;
begin
 read(j,k,l);
 if (j+k<l) or (k+l<j) or (j+l<k) then write('NO')
 else if (j=k) and (k=l)then write('Equilateral')
 else if (j*j+k*k=l*l) or (k*k+l*l=j*j) or (j*j+l*l=k*k) then write('Right')
 else write('General');
end.